### ISSUE TITLE
Fails to run on Ubuntu 18.04 LTS

### ISSUE THREAD

**Issue by dypa** [k0kubun/xremap#101]
CI pipeline generates binary file that not compatible with Ubuntu 18.04 LTS

steps to reproduce problem in docker
```
➜  ~ docker pull ubuntu:18.04 && docker run -it ubuntu:18.04

root@1e5455d3f0bf:/# apt update && apt install -y unzip wget libx11-dev
root@1e5455d3f0bf:/# wget https://github.com/k0kubun/xremap/releases/download/v0.4.2/xremap-linux-x86_64-x11.zip
root@1e5455d3f0bf:/# unzip xremap-linux-x86_64-x11.zip
root@1e5455d3f0bf:/# ./xremap 
./xremap: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.29' not found (required by ./xremap)
./xremap: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by ./xremap
```

possible solution - change in https://github.com/k0kubun/xremap/blob/master/.github/workflows/build.yml `runs-on: ubuntu-latest` to `runs-on: ubuntu-18.04`

**Comment by k0kubun** [comment]
Thanks, good catch. I'll also see if we can compile it with musl-libc to drop the dynamic dependency of libc.

**Comment by k0kubun** [comment]
I tried using cross-rs to solve musl integration and Arm support at once https://github.com/k0kubun/xremap/pull/104, but it seems like I need to change the X11 client first https://github.com/k0kubun/xremap/issues/92. So I just changed the CI config to use `ubuntu-18.04` for now as you suggested. Please try v0.4.3.

**Comment by dypa** [comment]
tested v0.4.3:
* ubuntu 18.04 - now works, fixed
* ubuntu 20.04 - still works, no regression
* debian 11 - still works, no regression
* debian 10 - now works, fixed

**Comment by k0kubun** [comment]
For the record, I implemented the above improvement in v0.5.0, meaning it statically links musl-libc and doesn't depend on glibc anymore. So you'll less likely to see the issue again.
